### PR TITLE
State machine for rtu over tcp

### PR DESCRIPTION
**Pull request by wesseloosthuizen**
Implement state machine for the rtu over tcp variant of the modbus protocol

### CODE REVIEW

File: examples/imqs-timeseries-modbus-site.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "name": "Site1",
     "address": "127.0.0.1",
     "controllers": [{
+        "endianness" : "BIG",
```
Comment (JacoBriers) [inline examples/imqs-timeseries-modbus-site.json]: We can move this to member of controller type at some point

File: src/github.com/goburrow/modbus/rtuovertcpclient.go [hunk 2]
```diff
@@ -18,3 +25,242 @@ func RTUoverTCPClient(address string) Client {
...
+		num, err = mb.conn.Read(reply)
+		// Typecast once
+		cnt = byte(num)
+		if err != nil {
+		} else {
+			if num == 0 {
+			} else {
+				var n byte
+				in = 0
+				fmt.Println("STATE")
```
Comment (JacoBriers) [inline src/github.com/goburrow/modbus/rtuovertcpclient.go]: Rather `mb.Logger.Printf(` ? Or is this how they did it in 3rd party code?

File: src/github.com/goburrow/modbus/rtuovertcpclient.go [hunk 3]
```diff
@@ -18,3 +25,242 @@ func RTUoverTCPClient(address string) Client {
...
+
+const (
+	lSID StateLength = 1
+	lFNC StateLength = 1
+	lLEN StateLength = 1
+	lEXC StateLength = 1
+	lCRC StateLength = 2
+	lDAT StateLength = 250 // Maximum possible data portion in response
+)
+
```
Comment (LeonLeibbrandt) [inline src/github.com/goburrow/modbus/rtuovertcpclient.go]: Above two const definition a bit cryptic, please add explanation in comment at one them, also const names usually start with capital letter.

**Comment by LeonLeibbrandt** [comment]
Looks good.

File: src/github.com/goburrow/modbus/rtuovertcpclient.go [hunk 4]
```diff
@@ -18,3 +25,242 @@ func RTUoverTCPClient(address string) Client {
...
+	reply := make([]byte, lSID+lFNC+lLEN+lDAT+lCRC)
+	for err == nil {
+		// Clear buffer
+		mb.flush(reply)
+		num, err = mb.conn.Read(reply)
+		// Typecast once
+		cnt = byte(num)
+		if err != nil {
+		} else {
+			if num == 0 {
```
Comment (JacoBriers) [inline src/github.com/goburrow/modbus/rtuovertcpclient.go]: `if num != 0 {`

**Pull request merged**